packages/live-share-inking/src/core/SharedInkingSession.ts

```diff
@@ -0,0 +1,695 @@
+            <svg width="${cursorSize}" height="${cursorSize}">
+                <path d="M0 0 L${cursorSize} ${cursorSize / 2.2} L${cursorSize / 2} ${cursorSize / 2} L8 ${cursorSize} Z" stroke-width="1" stroke="${borderColor}" fill="${backgroundColor}"/>
+            </svg>`;
+
+        if (this.userInfo) {
+            if (this.userInfo.displayName) {
+                visualTemplate += `
+                    <div style="background-color: ${backgroundColor}; color: ${foregroundColor}; border: 1px solid ${borderColor};
+                    border-radius: ${cursorSize * 1.1 / 2}px / 50%; padding: 2px 6px; margin: ${cursorSize * 0.75}px 0 0 -${cursorSize * 0.25}px; white-space: nowrap; font-size: 12px; font-family: sans-serif">
+                        ${this.userInfo.displayName}
+                    </div>`;
+            }
+            else if (this.userInfo.pictureUri && !this.userInfo.displayName) {
+                visualTemplate += `
+                    <img src="${this.userInfo.pictureUri}" style="width: ${cursorSize * 1.1}px; height: ${cursorSize * 1.1}px; border-radius: 50%;
+                        border: 1px solid ${borderColor}; margin: ${cursorSize * 0.75}px 0 0 -${cursorSize * 0.25}px;">`;
+            }
+            else if (this.userInfo.pictureUri && this.userInfo.displayName) {
+                visualTemplate += `
+                    <div style="display: flex; flex-direction: row; align-items: center; background-color: ${backgroundColor}; color: ${foregroundColor}; border: 1px solid ${borderColor};
+                        border-radius: ${cursorSize * 1.1 / 2}px / 50%; margin: ${cursorSize * 0.75}px 0 0 -${cursorSize * 0.25}px; white-space: nowrap; font-size: 12px; font-family: sans-serif">
+                        <img src="${this.userInfo.pictureUri}" style="width: ${cursorSize * 1.1}px; height: ${cursorSize * 1.1}px; border-radius: 50%;">
+                        <div style="padding: 0 8px">${this.userInfo.displayName}</div>
+                    </div>`;
+            }
+        }
+
+        const template = document.createElement("template");
+        template.innerHTML = visualTemplate;
+
+        const element = document.createElement("div");
+        element.style.position = "absolute";
+        element.style.display = "flex";
+        element.style.flexDirection = "row";
+
+        element.appendChild(template.content.cloneNode(true));
+
+        return element;
+    }
+
+    constructor(public clientId: string, public readonly userInfo?: IUserInfo) {
+        super(clientId, userInfo);
+
+        this._color = BuiltInCursorVisual.cursorColors[BuiltInCursorVisual.currentColorIndex];
+
+        BuiltInCursorVisual.currentColorIndex++;
+
+        if (BuiltInCursorVisual.currentColorIndex >= BuiltInCursorVisual.cursorColors.length) {
+            BuiltInCursorVisual.currentColorIndex = 0;
+        }
+    }
+}
+
+/**
+ * Enables live and collaborative inking.
+ */
+export class SharedInkingSession extends DataObject {
```

Comment:
We should discuss the naming of this class and whether we want to adopt the naming of the other DDS's in the SDK. We're talking about changing everything with `Ephemeral***` to `Live***` so maybe this class should be `LiveInkingSession`.